pkg/fission-cli/cmd/client.go

```diff
@@ -101,30 +102,43 @@ func NewClient(opts ClientOptions) (*Client, error) {
 	client := &Client{}
 	var err error
 	var cmdConfig clientcmd.ClientConfig
+	var restConfig *rest.Config
 	if len(opts.Namespace) > 0 {
 		client.Namespace = opts.Namespace
 	} else {
 		cmdConfig, err = GetClientConfig(opts.KubeContext)
 		if err != nil {
-			return nil, err
+			console.Verbose(2, err.Error())
 		}
-		namespace, _, err := cmdConfig.Namespace()
-		if err != nil {
-			return nil, err
+
+		if cmdConfig != nil {
+			namespace, _, err := cmdConfig.Namespace()
+			if err != nil {
+				return nil, err
+			}
+			client.Namespace = namespace
+			console.Verbose(2, "Kubeconfig default namespace %q", namespace)
 		}
-		client.Namespace = namespace
 	}
 
-	console.Verbose(2, "Kubeconfig default namespace %q", client.Namespace)
-
 	if opts.RestConfig != nil {
 		client.RestConfig = opts.RestConfig
-	} else {
+	} else if cmdConfig != nil {
 		restConfig, err := cmdConfig.ClientConfig()
 		if err != nil {
 			return nil, err
 		}
 		client.RestConfig = restConfig
+	} else {
+		console.Verbose(2, "Using in-cluster config")
+		restConfig, err = rest.InClusterConfig()
```

Comment:
We can make this local definition and remove above  definition.

```
restConfig, err := rest.InClusterConfig()
```
remove above `var restConfig *rest.Config`